tests/flytekit/unit/types/structured_dataset/test_structured_dataset_workflow_with_nested_type.py

```diff
@@ -0,0 +1,297 @@
+import os
+import typing
+from dataclasses import dataclass
+
+import numpy as np
+import pyarrow as pa
+import pyarrow.parquet as pq
+import pytest
+from typing_extensions import Annotated
+
+from flytekit import FlyteContext, FlyteContextManager, StructuredDataset, kwtypes, task, workflow
+from flytekit.models import literals
+from flytekit.models.literals import StructuredDatasetMetadata
+from flytekit.models.types import StructuredDatasetType
+from flytekit.types.structured.basic_dfs import CSVToPandasDecodingHandler, PandasToCSVEncodingHandler
+from flytekit.types.structured.structured_dataset import (
+    DF,
+    PARQUET,
+    StructuredDatasetDecoder,
+    StructuredDatasetEncoder,
+    StructuredDatasetTransformerEngine,
+)
+
+pd = pytest.importorskip("pandas")
+
+PANDAS_PATH = FlyteContextManager.current_context().file_access.get_random_local_directory()
+NUMPY_PATH = FlyteContextManager.current_context().file_access.get_random_local_directory()
+BQ_PATH = "bq://flyte-dataset:flyte.table"
+
+
+@dataclass
+class MyCols:
+    Name: str
+    Age: int
+
+
+my_cols = kwtypes(Name=str, Age=int)
+my_dataclass_cols = MyCols
+my_dict_cols = {"Name": str, "Age": int}
+fields = [("Name", pa.string()), ("Age", pa.int32())]
+arrow_schema = pa.schema(fields)
+pd_df = pd.DataFrame({"Name": ["Tom", "Joseph"], "Age": [20, 22]})
+
+
+class MockBQEncodingHandlers(StructuredDatasetEncoder):
+    def __init__(self):
+        super().__init__(pd.DataFrame, "bq", "")
+
+    def encode(
+        self,
+        ctx: FlyteContext,
+        structured_dataset: StructuredDataset,
+        structured_dataset_type: StructuredDatasetType,
+    ) -> literals.StructuredDataset:
+        return literals.StructuredDataset(
+            uri="bq://bucket/key", metadata=StructuredDatasetMetadata(structured_dataset_type)
+        )
+
+
+class MockBQDecodingHandlers(StructuredDatasetDecoder):
```

Comment:
ditto

Reply:
I'd better just remove the useless  redundant code (`MockBQEncodingHandlers `, `MockBQDecodingHandlers`).